src/services/tx/__tests__/txSender.test.ts

```diff
@@ -130,6 +144,40 @@ describe('txSender', () => {
 
       expect(proposeTx).toHaveBeenCalledWith('4', '0x123', '0x456', tx, '0x1234567890')
       expect(proposedTx).toEqual({ txId: '123' })
+
+      expect(txEvents.txDispatch).toHaveBeenCalledWith('PROPOSED', { txId: '123' })
+    })
+
+    it('should dispatch a tx proposal with a signature', async () => {
+      const tx = await createTx({
+        to: '0x123',
+        value: '1',
+        data: '0x0',
+      })
+
+      const proposedTx = await dispatchTxProposal('4', '0x123', '0x456', tx, '345')
+
+      expect(proposeTx).toHaveBeenCalledWith('4', '0x123', '0x456', tx, '0x1234567890')
+      expect(proposedTx).toEqual({ txId: '123' })
+
+      expect(txEvents.txDispatch).toHaveBeenCalledWith('SIGNATURE_PROPOSED', { txId: '123' })
+    })
+
+    it('should fail to propose', async () => {
```

Comment:
We should also test when new transactions fail to propose.

Reply:
Added 👍